Source/UI/IdentityUI.swift

```diff
@@ -300,27 +302,64 @@ public class IdentityUI {
             uiResult = .failed(error)
         }
 
-        // This is no more the currently presented login flow.
-        IdentityUI.presentedIdentityUI = nil
-
         let finish = { [weak self] in
+            if case let .completed(user)? = uiResult, shouldPersistUser {
+                user.persistCurrentTokens()
+            }
             if let result = uiResult {
                 self?.delegate?.didFinish(result: result)
             }
         }
 
-        if self.navigationController.presentingViewController != nil {
-            // It might be that `IdentityUIViewController.endLoading()` has been called just before getting here, in case the result of a networking
-            // operation caused the flow to end. The `endLoading()` method will then trigger a `view.isUserInteractionEnabled = true`, which would cause the
-            // keyboard to show up again during the dismiss animation, resulting in a very weird and funky UI glitch. In order to avoid that, we force an
-            // `endEditing()` on the topmost view (if any) before starting the view dismissing.
-            self.navigationController.topViewController?.view.endEditing(true)
+        let dismissFlow = { [weak self] in
+
+            // This is no more the currently presented login flow
+            IdentityUI.presentedIdentityUI = nil
+
+            if self?.navigationController.presentingViewController != nil {
+                // It might be that `IdentityUIViewController.endLoading()` has been called just before getting here, in case the result of a networking
+                // operation caused the flow to end. The `endLoading()` method will then trigger a `view.isUserInteractionEnabled = true`, which would cause the
+                // keyboard to show up again during the dismiss animation, resulting in a very weird and funky UI glitch. In order to avoid that, we force an
+                // `endEditing()` on the topmost view (if any) before starting the view dismissing.
+                self?.navigationController.topViewController?.view.endEditing(true)
```

Comment:
Is this an issue that comes from some other kind of bug or issue? And if it is, should that be fixed instead?

Reply:
Ya, perhaps, this should be looked in to, @lrossi had the original code here so maybe he has some insight?